ssl/statem/statem_clnt.c

```diff
@@ -1362,7 +1364,22 @@ MSG_PROCESS_RETURN dtls_process_hello_verify(SSL_CONNECTION *s, PACKET *pkt)
         return MSG_PROCESS_ERROR;
     }
     s->d1->cookie_len = cookie_len;
+    if (ssl == NULL) {
+        SSLfatal(s, SSL_AD_DECODE_ERROR, SSL_R_LENGTH_MISMATCH);
```

Comment:
This is an internal error - SSL_R_LENGTH_MISMATCH and SSL_AD_DECODE_ERROR seem wrong to me.